src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -288,7 +290,14 @@ private static PoisonedFileEntry CheckSingleFile(IEnumerable<CatalogPackageEntry
                 AssemblyName asm = AssemblyName.GetAssemblyName(fileToCheck);
                 if (IsAssemblyPoisoned(fileToCheck))
                 {
-                    poisonEntry.Type |= PoisonType.AssemblyAttribute;
+                    if (fileToCheck.EndsWith(".dll") && HasAttributeOfType(fileToCheck, "System.Runtime.CompilerServices.ReferenceAssemblyAttribute"))
```

Comment:
Because this is within the IsAssemblyPoisoned check, it implies to me this won't work for any crossgen'd assemblies.  Is that correct?

Reply:
That is correct. In order to assign a poison type, it needs to be validated that the assembly has been poisoned first, which currently involves checking for the "POSIONED" marker. Given that crossgen is altering the files and removing this marker, the check for IsAssemblyPoisoned will not work on crossgen'd assemblies.

Once https://github.com/dotnet/source-build/issues/2804 is addressed, it will be possible to move this attribute assignment out of the IsAssemblyPoisoned logic. That is, the IsAssemblyPoisoned logic can be adapted to check for the SBRP attribute and assign a PoisonType based on that.

Reply:
My thought are that we should add the SBRP marker first.  My reasoning is two fold.  The proposed changes aren't able to detect cross gen SBRP leaks and the logic that poisons the SBRPs isn't needed once the SBRP marker is in place.  Getting the SBRP marker in place up front be a better solution all around.

Reply:
Updated in https://github.com/dotnet/installer/pull/17339/commits/7e4372eb95f9bbc49cbd9db34159c9332c68f52c. Working to flow the SBRP attribute changes into this.